src/drive/storage/index.ts

```diff
@@ -0,0 +1,244 @@
+        parentFolderId: payload.parentFolderId,
+        folderName: payload.folderName,
+      }, {
+        cancelToken: cancelTokenSource.token,
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      });
+
+    return [promise, cancelTokenSource];
+  }
+
+  /**
+   * Moves a specific folder to a new location
+   * @param payload
+   */
+  public async moveFolder(payload: MoveFolderPayload): Promise<MoveFolderResponse> {
+    return this.axios
+      .post(`${this.apiUrl}/api/storage/move/folder`, {
+        folderId: payload.folder.id,
+        destination: payload.destinationFolderId,
+      }, {
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      });
+  }
+
+  /**
+   * Updates the metadata of a folder
+   * @param payload
+   */
+  public async updateFolder(payload: UpdateFolderMetadataPayload): Promise<void> {
+    await this.axios
+      .post(`${this.apiUrl}/api/storage/folder/${payload.folderId}/meta`, {
+        metadata: payload.changes
+      }, {
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      });
+  }
+
+  /**
+   * Fetches & returns the contents of a specific folder
+   * @param folderId
+   */
+  public getFolderContent(folderId: number): [
+    Promise<{
+      folders: DriveFolderData[];
+      files: DriveFileData[]
+    }>,
+    CancelTokenSource
+  ] {
+    const cancelTokenSource = axios.CancelToken.source();
+    const promise = this.axios
+      .get<FetchFolderContentResponse>(`${this.apiUrl}/api/storage/v2/folder/${folderId}`, {
+        cancelToken: cancelTokenSource.token,
+        headers: this.headers()
+      })
+      .then(response => {
+        return {
+          folders: response.data.children.map(folder => ({ ...folder, isFolder: true })),
+          files: response.data.files,
+        };
+      });
+
+    return [promise, cancelTokenSource];
+  }
+
+  /**
+   * Removes a specific folder from the centralized persistence
+   * @param folderId
+   */
+  public deleteFolder(folderId: number): Promise<unknown> {
+    return this.axios
+      .delete(`${this.apiUrl}/api/storage/folder/${folderId}`, {
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      });
+  }
+
+  /**
+   * Creates a new file entry
+   * @param fileEntry
+   */
+  public createFileEntry(fileEntry: FileEntry): Promise<DriveFileData> {
+    return this.axios
+      .post(`${this.apiUrl}/api/storage/file`, {
+        file: {
+          fileId: fileEntry.id,
+          type: fileEntry.type,
+          bucket: fileEntry.bucket,
+          size: fileEntry.size,
+          folder_id: fileEntry.folder_id,
+          name: fileEntry.name,
+          encrypt_version: fileEntry.encrypt_version,
+        }
+      }, {
+        headers: this.headers()
+      })
+      .then(response => {
+        if (response.status === 402) {
```

Comment:
Maybe using a middleware with status codes that are frequent like 401, 402, 403, 500 is better than dealing with them for each response

Reply:
I wouldn't generalize so soon.. the only request that was checking for this error was that one, and checking the server I see no endpoints sending this error code.
I added tho an AppError class so we can return the status code to the client so it can show the errors depending on the codes. I saw this is already happening.
Once we start adding validations and error management we can see which code we use and maybe add a middleware.
Wdyt?